agent/otel/otlpmqttexporter/otlp.go

```diff
@@ -196,6 +196,13 @@ func (e *baseExporter) injectScopeLogsAttribute(logsScope plog.ScopeLogs, attrib
 	return logsScope
 }
 
+func (e *baseExporter) shutdown(_ context.Context) error {
+	if e.config.Client == nil || !(*e.config.Client).IsConnected() {
+		(*e.config.Client).Disconnect(0)
```

Comment:
Won't it panic if `e.config.Client` is `nil`?

Reply:
Also, why would we need to disconnect if it's not connected?

Reply:
By Disconnecting the handler of disconnection will kick in, and do the retry to connect. And yeah, the logic was inverted there.